text/tls-and-alpn.rst

```diff
@@ -0,0 +1,380 @@
+TLS and ALPN
+============
+
+This RFC proposes to change the transport of EdgeDB frontend connections to TLS,
+and use ALPN for multiplexing protocol selection.
+
+
+Motivation
+----------
+
+The Transport Layer Security (TLS) Protocol is widely used for providing privacy
+and data integrity between two communicating applications [1]_. EdgeDB as a
+database should naturally support TLS as the transport layer protocol.
+
+With the help of EdgeDB tooling, it will be easy to setup TLS certificates for
+both development and production environments. Given also that the benchmark
+result of EdgeDB throughput overhead comparing TLS to vanilla TCP is about
+5% ~ 15%, it is also proposed to communicate in TLS by default.
+
+.. image:: ./EdgeDB-TLS-Benchmark.png
+
+At last, with TLS enabled by default, it is possible to leverage the TLS
+Application-Layer Protocol Negotiation (ALPN) Extension [2]_ for secure and
+reliable protocol selection on top of the TLS transport, allowing EdgeDB to
+multiplex different frontend protocols like the binary protocol, EdgeQL over
+HTTP or GraphQL over HTTP (and maybe more in the futre) on the same port as
+recommended in Pre-RFC 1910 [3]_.
+
+
+Overview
+--------
+
+EdgeDB server CLI (``edb server``) will take a few more parameters to configure
+a proper TLS server::
+
+    --tls-certfile
```

Comment:
Perhaps if no `--tls-certfile` was provided we always auto-generate a local one?

Reply:
Yes that's the idea. I'll clarify it.